react-validated-input/my-app/src/Password.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+
+export function Password() {
+  const [input, setInput] = useState('');
+  const errorMessage = [];
+  if (input === '') {
+    errorMessage.push('A password is required.');
+  } else {
+    if (input.length < 8) {
+      errorMessage.push('Your password is too short');
+    }
+    if (!input.match(/\d/)) {
+      errorMessage.push('Your password lacks a number.');
+    }
+    if (!input.match(/[A-Z]/)) {
+      errorMessage.push('Your password lacks a capital letter.');
+    }
+    if (!input.match(/!|@|#|\$|%|\^|&|\(|\)/)) {
+      errorMessage.push('Your password lacks a special character.');
+    }
+  }
+  const symbolClassNames =
+    errorMessage.length === 0
+      ? 'fa-solid fa-check text-green-500'
+      : 'fa-solid fa-x text-red-500';
+  return (
+    <label htmlFor="password-input" className="font-semibold">
+      Password
+      <br />
+      <input
+        id="password-input"
```

Comment:
Why is your cohort so keen to style using the `id`? I've almost never seen that in React code and no prior cohort did this. It must be something Brandon taught you?

Reply:
I don't think I did use id for styling in this exercise. I used it to give the 'for' a target. FreeCodeCamp recommends both couching your inputs inside of labels and connecting them via for & id. As for styling via id in prior exercises, when a single element needs specific styling I guess we could give it a unique class name, I'm not sure why there might be a tendency to use id instead. Brandon didn't tell us to.